src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -428,12 +430,16 @@ private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace,
         return matlabLauncher.join();
     }
 
-    public List<String> constructMatlabCommandWithBatch() {
+    public List<String> constructMatlabCommandWithBatch() throws IOException, InterruptedException {
```

Comment:
It seems odd that a function called constructMatlabCommandWithBatch should throw an IOException. This looks like the exception specification of this method is determined by the implementation details of this method. 

Similarly, methods whose job is to construct commands (based on -r or -batch) is now tasked with the additional responsibility of doing copy of the scratch file. That seems out of place as well.